design/KruizeLocalAPI.md

```diff
@@ -0,0 +1,1986 @@
+      "namespace": "default",
+      "containers": [
+        {
+          "container_image_name": "kruize/tfb-db:1.15",
+          "container_name": "tfb-server-1"
+        }
+      ]
+    }
+  ]
+}
+```
+
+</details>
+
+---
+<a name="generate-recommendations-api"></a>
+
+### Generate Recommendations API
+
+**Note: This API is specific to the Local Monitoring use case.** <br>
+Generates the recommendation for a specific experiment based on provided parameters similar to update recommendations API.
+This can be called directly after creating the experiment and doesn't require the update results API as metrics are
+fetched from the provided `datasource` (E.g. Prometheus) instead of the database.
+
+**Request Parameters**
+
+| Parameter           | Type   | Required | Description                                                                                                                                |
+|---------------------|--------|----------|--------------------------------------------------------------------------------------------------------------------------------------------|
+| experiment_name     | string | Yes      | The name of the experiment.                                                                                                                |
+| interval_end_time   | string | optional | The end time of the interval in the format `yyyy-MM-ddTHH:mm:sssZ`. This should be the date on which recommendation needs to be generated. |
+| interval_start_time | string | optional | The start time of the interval in the format `yyyy-MM-ddTHH:mm:sssZ`.                                                                      |
+
+The recommendation API requires only one mandatory field i.e. `experiment_name`. Other optional parameter like `interval_end_time` will be fetched from the provided datasource.
+Similarly, `interval_start_time` will be calculated based on `interval_end_time`, if not provided. By utilizing
+these parameters, the API generates recommendations based on short-term, medium-term, and long-term factors. For
+instance, if the long-term setting is configured for `15 days` and the interval_end_time is set to `Jan 15 2023 00:00:
+00.000Z`, the API retrieves data from the past 15 days, starting from January 1st. Using this data, the API generates
+three recommendations for `Jan 15th 2023`.
+
+It is important to ensure that the difference between `interval_end_time` and `interval_start_time` should not exceed 15
+days. This restriction is in place to prevent potential timeouts, as generating recommendations beyond this threshold
+might require more time.
+
+**Request**
+
+`POST /generateRecommendations?experiment_name=?&interval_end_time=?`
+
+`POST /generateRecommendations?experiment_name=?&interval_end_time=?&interval_start_time=?`
+
+example
+
+`curl --location --request POST 'http://127.0.0.1:8080/generateRecommendations?interval_end_time=2023-01-02T00:15:00.000Z&experiment_name=temp_1'`
```

Comment:
Better to provide an example without the interval_end_time. Also use ` http://<URL>:<PORT>/` insead of `http://127.0.0.1:8080`

Reply:
Please explain what is returned when `interval_end_time` is not specified

Reply:
Included an example for case where `interval_end_time` is not specified